common/src/lib.rs

```diff
@@ -50,3 +50,24 @@ macro_rules! generate_logging_api {
         );
     };
 }
+
+/// A type that allows adding file and line numbers to log messages
+/// automatically. It should be instantiated at the root logger of each
+/// executable that desires this functionality, as in the following example.
+/// ```
+///     slog::Logger::root(drain, o!(FileLocationKv))
+/// ```
+pub struct FileKv;
+
+impl slog::KV for FileKv {
+    fn serialize(
+        &self,
+        record: &slog::Record,
+        serializer: &mut dyn slog::Serializer,
+    ) -> slog::Result {
```

Comment:
Does it make sense to limit this to `record.level() <= Level::Warning` to limit log bloat? There's a bunch of periodic `INFO`/`DEBUG` logs I think wouldn't really benefit from the file/line being included everytime?

Reply:
I think the info ones in particular are helpful to have line numbers, but I wouldn't mind turning this off for debug logs if necessary. Seems somewhat like a premature optimization, but I could totally see us rotating out useful information early just to keep line numbers when not needed, so I think the compromise is useful.

The reason I really wanted these is that a lot of times logs aren't unique, so having the info ones are particularly helpful. Debug logs can always add extra context.

Reply:
I filtered at info level in [0cf4013](https://github.com/oxidecomputer/omicron/pull/3782/commits/0cf4013d0705b61896c7675598e3b589c79ee3ff)